nifi-nar-bundles/nifi-standard-services/nifi-hbase_1_1_2-client-service-bundle/nifi-hbase_1_1_2-client-service/src/main/java/org/apache/nifi/hbase/HBase_1_1_2_LookupService.java

```diff
@@ -0,0 +1,180 @@
+            .build();
+    public static final PropertyDescriptor RETURN_QFS = new PropertyDescriptor.Builder()
+            .name("hb-lu-return-qfs")
+            .displayName("Column Qualifiers")
+            .description("The column qualifies that will be returned.")
+            .required(false)
+            .addValidator(Validator.VALID)
+            .build();
+    protected static final PropertyDescriptor CHARSET = new PropertyDescriptor.Builder()
+            .name("hb-lu-charset")
+            .displayName("Character Set")
+            .description("Specifies the character set of the document data.")
+            .required(true)
+            .defaultValue("UTF-8")
+            .addValidator(StandardValidators.CHARACTER_SET_VALIDATOR)
+            .build();
+
+    private String tableName;
+    private List<byte[]> families;
+    private List<byte[]> qualifiers;
+    private Charset charset;
+    private List<PropertyDescriptor> lookupProperties;
+
+    @Override
+    protected void init(ControllerServiceInitializationContext config) throws InitializationException {
+        super.init(config);
+        this.lookupProperties = new ArrayList<>();
+        this.lookupProperties.addAll(properties);
+        this.lookupProperties.add(TABLE_NAME);
+        this.lookupProperties.add(RETURN_CFS);
+        this.lookupProperties.add(RETURN_QFS);
+        this.lookupProperties.add(CHARSET);
+    }
+
+    @Override
+    public Optional<Record> lookup(Map<String, String> coordinates) throws LookupFailureException {
+        byte[] rowKey = ((String)coordinates.get("rowKey")).getBytes();
+        try {
+            Map<String, Object> values = new HashMap<>();
+            try (Table table = this.connection.getTable(TableName.valueOf(tableName))) {
+                Get get = new Get(rowKey);
+                Result result = table.get(get);
+
+                for (byte[] fam : families) {
+                    NavigableMap<byte[], byte[]>  map = result.getFamilyMap(fam);
+                    for (Map.Entry<byte[], byte[]> entry : map.entrySet()) {
+                        if (qualifiers.contains(entry.getKey()) || qualifiers.size() == 0) {
+                            values.put(new String(entry.getKey(), charset), new String(entry.getValue(), charset));
+                        }
+                    }
+                }
+            }
+
+            if (values.size() > 0) {
+                final List<RecordField> fields = new ArrayList<>();
+                for (String key : values.keySet()) {
+                    fields.add(new RecordField(key, RecordFieldType.STRING.getDataType()));
+                }
+                final RecordSchema schema = new SimpleRecordSchema(fields);
+                return Optional.ofNullable(new MapRecord(schema, values));
+            } else {
+                throw new LookupFailureException(String.format("Nothing was found that matched the criteria for row key %s", coordinates.get("rowKey")));
```

Comment:
Looks like Mark P had suggest returning Optional.empty() here instead of throwing an exception